ray-operator/controllers/ray/common/pod.go

```diff
@@ -72,22 +72,29 @@ func isOverwriteRayContainerCmd(instance rayv1.RayCluster) bool {
 	return ok && strings.ToLower(v) == "true"
 }
 
-func initTemplateAnnotations(instance rayv1.RayCluster, podTemplate *corev1.PodTemplateSpec) {
+func initTemplateAnnotations(instance *rayv1.RayCluster, podTemplate *corev1.PodTemplateSpec) {
 	if podTemplate.Annotations == nil {
 		podTemplate.Annotations = make(map[string]string)
 	}
-
+	// Validation for invalid case: FT explicitly enabled and GcsFaultToleranceOptions is not nil
+	if instance.Annotations[utils.RayFTEnabledAnnotationKey] == "false" && instance.Spec.GcsFaultToleranceOptions != nil {
```

Comment:
can you add a function `validateRayClusterSpec` and add the check in the very beginning of `rayClusterReconcile`?

You can refer to the following links:

https://github.com/ray-project/kuberay/blob/9068102246eeb5ab9d9e0b9a7480618d3f348686/ray-operator/controllers/ray/raycluster_controller.go#L232

https://github.com/ray-project/kuberay/blob/9068102246eeb5ab9d9e0b9a7480618d3f348686/ray-operator/controllers/ray/rayjob_controller.go#L132